src/sources/docker_logs.rs

```diff
@@ -596,18 +601,24 @@ impl EventStreamBuilder {
                                 emit!(DockerLogsLoggingDriverUnsupported {
                                     error,
                                     container_id: info.id.as_str(),
-                                })
+                                });
+                                Err(Persistence::Permanent)
+                            }
+                            _ => {
+                                emit!(DockerLogsCommunicationError {
+                                    error,
+                                    container_id: Some(info.id.as_str())
+                                });
+                                Err(Persistence::Transient)
                             }
-                            _ => emit!(DockerLogsCommunicationError {
-                                error,
-                                container_id: Some(info.id.as_str())
-                            }),
-                        };
-                        Err(())
+                        }
                     }
                 }
             })
-            .take_while(|v| ready(v.is_ok()))
+            .take_while(|v| {
+                error = v.as_ref().err().cloned();
+                ready(v.is_ok())
+            })
```

Comment:
Any chance of using something like `.last` to get the final error value instead of potentially repeatedly setting `error`?

Reply:
Not that I see. But the current code should set it max once.